specs/WebResourceResponseReceived.md

```diff
@@ -0,0 +1,219 @@
+    /// the associated request has fired. Modifications made to the request or response
+    /// objects are ignored.
+    interface ICoreWebView2WebResourceResponseReceivedEventHandler : IUnknown
+    {
+        /// Called to provide the implementer with the event args for the
+        /// corresponding event.
+        HRESULT Invoke(
+            [in] ICoreWebView2* sender,
+            [in] ICoreWebView2WebResourceResponseReceivedEventArgs* args);
+    }
+
+    /// Completion handler for PopulateResponseContent async method. It's invoked
+    /// when the Content stream of the Response of a WebResourceResponseReceieved
+    /// event is available.
+    interface ICoreWebView2WebResourceResponseReceivedEventArgsPopulateResponseContentCompletedHandler : IUnknown
+    {
+        /// Called to provide the implementer with the completion status
+        /// of the corresponding asynchronous method call.
+        HRESULT Invoke([in] HRESULT errorCode);
+    }
+
+    /// Event args for the WebResourceResponseReceived event. Will contain the
+    /// request as it was sent and the response as it was received.
+    /// Note: To get the response content stream, first call PopulateResponseContent
+    /// and wait for the async call to complete, otherwise the content stream object
+    /// returned will be null.
+    interface ICoreWebView2WebResourceResponseReceivedEventArgs : IUnknown
+    {
+        /// Web resource request object. Any modifications to this object will be ignored.
+        [propget] HRESULT Request([out, retval] ICoreWebView2WebResourceRequest** request);
+        /// Web resource response object. Any modifications to this object
+        /// will be ignored.
+        [propget] HRESULT Response([out, retval] ICoreWebView2WebResourceResponse** response);
+
+        /// Async method to request the Content stream of the response.
+        HRESULT PopulateResponseContent(ICoreWebView2WebResourceResponseReceivedEventArgsPopulateResponseContentCompletedHandler* handler);
+    }
+}
+```
+
+## WinRT
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+    runtimeclass CoreWebView2
+    {
+        // ...
+
+        /// WebResourceResponseReceived event fires after the WebView has received and processed the response for a WebResource request.
+        event Windows.Foundation.TypedEventHandler<CoreWebView2, CoreWebView2WebResourceResponseReceivedEventArgs> WebResourceResponseReceived;
+    }
+
+    /// Event args for the WebResourceResponseReceived event.
+    runtimeclass CoreWebView2WebResourceResponseReceivedEventArgs
+    {
+        /// Web resource request object.
+        /// Any modifications to this object will be ignored.
+        CoreWebView2WebResourceRequest Request { get; };
+        /// Web resource response object.
+        /// Any modifications to this object will be ignored.
+        CoreWebView2WebResourceResponse Response { get; };
+
+        /// Async method to request the Content stream of the response.
```

Comment:
What happens if you call this method a second time?

Reply:
That's exactly what will happen. I updated to specify that.